tests/test_debian_9.py

```diff
@@ -814,38 +816,121 @@ def test_compose_2idx2sh(self):
+
+        assert rc == 0
+        # Wait for containers to be healthy
+        assert self.wait_for_containers(container_count)
+        # Check Splunkd on all the containers
+        assert self.check_splunkd("admin", self.password)
         # Check values in log output
-        self.check_common_keys(log_json_sh1, "sh")
-        self.check_common_keys(log_json_sh2, "sh")
-        self.check_common_keys(log_json_idx1, "idx")
-        self.check_common_keys(log_json_idx2, "idx")
+        log_json = {}
+        for c in container_names:
+            log_json[c] = self.extract_json(c)
+        self.check_common_keys(log_json["sh1"], "sh")
+        self.check_common_keys(log_json["sh2"], "sh")
+        self.check_common_keys(log_json["idx1"], "idx")
+        self.check_common_keys(log_json["idx2"], "idx")
+        self.check_common_keys(log_json["cm1"], "cm")
         try:
-            assert log_json_sh1["splunk_indexer"]["hosts"] == ["idx1", "idx2"]
-            assert log_json_sh1["splunk_search_head"]["hosts"] == ["sh1", "sh2"]
-            assert log_json_sh2["splunk_indexer"]["hosts"] == ["idx1", "idx2"]
-            assert log_json_sh2["splunk_search_head"]["hosts"] == ["sh1", "sh2"]
-            assert log_json_idx1["splunk_indexer"]["hosts"] == ["idx1", "idx2"]
-            assert log_json_idx1["splunk_search_head"]["hosts"] == ["sh1", "sh2"]
-            assert log_json_idx2["splunk_indexer"]["hosts"] == ["idx1", "idx2"]
-            assert log_json_idx2["splunk_search_head"]["hosts"] == ["sh1", "sh2"]
+            assert log_json["cm1"]["all"]["vars"]["splunk"]["indexer_cluster"] == True
+            for c in container_names:
+                assert log_json[c]["splunk_indexer"]["hosts"] == ["idx1", "idx2"]
+                assert log_json[c]["splunk_search_head"]["hosts"] == ["sh1", "sh2"]
+                assert log_json[c]["splunk_cluster_master"]["hosts"] == ["cm1"]
         except KeyError as e:
             self.logger.error(e)
             assert False
+
+        # Check connections
+        idx_list = ["idx1", "idx2"]
+        sh_list = ["sh1", "sh2", "cm1"]
+
+        containers = self.client.containers(filters={"label": "com.docker.compose.service={}".format("cm1")})
+        splunkd_port = self.client.port(containers[0]["Id"], 8089)[0]["HostPort"]
+
+        status, content = self.handle_request_retry("GET", "https://localhost:{}/services/cluster/master/searchheads?output_mode=json".format(splunkd_port), 
+                                                    {"auth": ("admin", self.password), "verify": False})
+        assert status == 200
+        output = json.loads(content)
+        for sh in output["entry"]:
+            if sh["content"]["label"] in sh_list and sh["content"]["status"] == "Connected":
+                sh_list.remove(sh["content"]["label"])
+
+        status, content = self.handle_request_retry("GET", "https://localhost:{}/services/cluster/master/peers?output_mode=json".format(splunkd_port), 
```

Comment:
careful doing "localhost" vs "loopback" (127.0.0.1).  They process differently and interface with interfaces differently.  If you don't expose on 0.0.0.0:port, this will fail.